providers/universal-provider/src/UniversalProvider.ts

```diff
@@ -231,7 +232,13 @@ export class UniversalProvider implements IUniversalProvider {
       // ignore without active session
       if (!this.session) return;
       const [namespace, chainId] = this.validateChain(chain);
-      this.getProvider(namespace).setDefaultChain(chainId, rpcUrl);
+      const provider = this.getProvider(namespace);
+      // @ts-expect-error
+      if (provider.name === "generic") {
```

Comment:
Why do we use CAIP2 format for the generic provider and jus the `chainId` in the other case ?
On another note, is this used elsewhere ? If so it might be better to use a constant.
```suggestion
      if (provider.name === GENERIC_PROVIDER_NAME) {
```

Reply:
>Why do we use CAIP2 format for the generic provider and jus the chainId in the other case ?

because custom providers set the `namespace` part of caip10 internally. This generic provider doesn't know any namespace and it actually supports multiple at the same time so it needs the whole caip2 as param unlike the others